File: web/backend/src/api/oauth/oauth_api.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from api.auth.schemas import AuthResponse
from api.deps import get_db, get_cache
from api.oauth.schemas import OAuthCallbackRequest
from api.oauth.service import get_auth_url, handle_callback
from core.decorators import handle_errors
from core.response import ApiResponse, ok
from core.cache import KeyValueCache

router = APIRouter(prefix="/api/oauth", tags=["oauth"])


@router.get("/auth-url")
@handle_errors
async def auth_url(
    provider: str,
    cache: KeyValueCache = Depends(get_cache),
):
    url = await get_auth_url(provider, cache)
    return ok({"authUrl": url})


@router.post("/callback", response_model=ApiResponse[AuthResponse])
@handle_errors
async def callback(
    req: OAuthCallbackRequest,
    db: AsyncSession = Depends(get_db),
    cache: KeyValueCache = Depends(get_cache),
):
    result = await handle_callback(req.provider, req.code, req.state, cache, db)
    return ok(result)

File: web/frontend/src/services/oauthApi.ts
import request from './request'
import type { ApiResponse } from '@/types/api'
import type { AuthResponse } from '@/types/auth'

export const oauthApi = {
  /** 获取第三方登录授权跳转 URL */
  getAuthUrl: (provider: string) =>
    request.get<ApiResponse<{ authUrl: string }>>('/oauth/auth-url', {
      params: { provider },
    }),

  /** 处理 OAuth 回调 */
  handleCallback: (provider: string, code: string, state: string) =>
    request.post<ApiResponse<AuthResponse>>('/oauth/callback', { provider, code, state }),
}
